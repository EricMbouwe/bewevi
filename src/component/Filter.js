import PropTypes from 'prop-types';
// import filterIcon from '../assets/filter.svg';
// import filterIcon from '../assets/filter.svg';

const Filter = ({
  handleChangeCategory,
  handleFilterName,
  category,
  filterName,
  categoryNameList,
  categoryList,
}) => (
  <div className="filter main">
    <select className="filter filter-select-category" name="filter-category" onChange={handleChangeCategory} value={category}>
      <option defaultValue="none" disabled hidden>
        Filter By Category
      </option>
      <option defaultValue="none">
        All
      </option>
      {categoryNameList.map(oneCategory => (
        <option key={oneCategory[0]} value={oneCategory[1]}>
          {oneCategory[0]}
        </option>
      ))}
    </select>
    {category !== 'Filter By Category' && category !== 'All' && (
      <select className="filter filter-select-name" name="filter-name" onChange={handleFilterName} value={filterName}>
        <option defaultValue="none" disabled hidden>
          Filter Name
        </option>
        <option defaultValue="none">
          All
        </option>
        {categoryList.map(oneName => (
          <option key={oneName} value={oneName}>
            {oneName.charAt(0).toUpperCase() + oneName.slice(1)}
          </option>
        ))}
      </select>
    )}
    <div className="filter filter-close-box" />

  </div>

);

Filter.propTypes = {
  handleChangeCategory: PropTypes.func.isRequired,
  categoryList:
    PropTypes.arrayOf(
      PropTypes.string.isRequired,
    ).isRequired,
  categoryNameList:
    PropTypes.arrayOf(
      PropTypes.arrayOf(
        PropTypes.string.isRequired,
      ).isRequired,
    ).isRequired,
  handleFilterName: PropTypes.func.isRequired,
  filterName: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
};

export default Filter;
