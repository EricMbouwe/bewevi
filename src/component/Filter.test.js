import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import Filter from './Filter';

const handleChangeCategory = jest.fn();
const handleFilterName = jest.fn();
let category = 'Filter By Category';
const filterName = 'Filter Name';
const categoryNameList = [
  ['Habitat', 'pokemon-habitat/'],
  ['Shape', 'pokemon-shape/'],
];
const categoryList = ['cave', 'forest'];
const renderReadyComponent = (
  <Filter
    handleChangeCategory={handleChangeCategory}
    handleFilterName={handleFilterName}
    category={category}
    filterName={filterName}
    categoryNameList={categoryNameList}
    categoryList={categoryList}
  />
);

describe('<Filter />', () => {
  describe('If category is \'Filter By Category\'', () => {
    it('renders category list, not filter name options ', () => {
      render(
        renderReadyComponent,
      );
      expect(screen.getByText(/Habitat/i)).toBeInTheDocument();
      expect(screen.getByText(/Shape/i)).toBeInTheDocument();
      expect(screen.getAllByText(/All/i).length).toEqual(1);
      expect(screen.queryByText(/Cave/i)).not.toBeInTheDocument();
    });

    it('renders correctly', () => {
      const tree = renderer.create(
        renderReadyComponent,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('If category is not \'Filter By Category\' or \'All\'', () => {
    it('renders category list, filter name options if category is not \'Filter By Category\' or \'All\'', () => {
      category = 'Shape';
      render(
        <Filter
          handleChangeCategory={handleChangeCategory}
          handleFilterName={handleFilterName}
          category={category}
          filterName={filterName}
          categoryNameList={categoryNameList}
          categoryList={categoryList}
        />,
      );
      expect(screen.getByText(/Habitat/i)).toBeInTheDocument();
      expect(screen.getByText(/Shape/i)).toBeInTheDocument();
      expect(screen.getAllByText(/All/i).length).toEqual(2);
      expect(screen.getByText(/Cave/i)).toBeInTheDocument();
    });

    it('renders correctly', () => {
      const tree = renderer.create(
        renderReadyComponent,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
