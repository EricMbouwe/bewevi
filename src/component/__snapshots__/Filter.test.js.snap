// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> If category is 'Filter By Category' renders correctly 1`] = `
<div
  className="filter main"
>
  <select
    className="filter filter-select-category"
    name="filter-category"
    onChange={[MockFunction]}
    value="Filter By Category"
  >
    <option
      defaultValue="none"
      disabled={true}
      hidden={true}
    >
      Filter By Category
    </option>
    <option
      defaultValue="none"
    >
      All
    </option>
    <option
      value="pokemon-habitat/"
    >
      Habitat
    </option>
    <option
      value="pokemon-shape/"
    >
      Shape
    </option>
  </select>
  <div
    className="filter filter-close-box"
  />
</div>
`;

exports[`<Filter /> If category is not 'Filter By Category' or 'All' renders correctly 1`] = `
<div
  className="filter main"
>
  <select
    className="filter filter-select-category"
    name="filter-category"
    onChange={[MockFunction]}
    value="Filter By Category"
  >
    <option
      defaultValue="none"
      disabled={true}
      hidden={true}
    >
      Filter By Category
    </option>
    <option
      defaultValue="none"
    >
      All
    </option>
    <option
      value="pokemon-habitat/"
    >
      Habitat
    </option>
    <option
      value="pokemon-shape/"
    >
      Shape
    </option>
  </select>
  <div
    className="filter filter-close-box"
  />
</div>
`;
